development/image-guard/image-guard/templates/deployment.yaml

```diff
@@ -0,0 +1,57 @@
+{{- $svcHost := printf "%s.%s.svc" .Release.Name .Release.Namespace -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.deployment.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}
+    spec:
+      containers:
+        - name: controller
+          image: {{ .Values.deployment.image }}
+          command: ["/image-guardd"]
+          env:
+            - name: HOST_NAME
+              value: {{ $svcHost }}
+          args:
+            - "--host"
+            - {{ $svcHost | quote }}
+            - "--cert-path"
+            - "/certs/cert.crt"
+            - "--key-path"
+            - "/certs/key.pem"
+            {{- with .Values.enforcedRegistry }}
+            {{- if .enabled }}
+            {{- range .allowedRegistries }}
+            - "--allowed-registry"
+            - {{ . | quote }}
+            {{- end }}{{ end }}{{ end }}
+          imagePullPolicy: {{ .Values.deployment.imagePullPolicy }}
+          ports:
+            - containerPort: 8443
+              name: server-tls-port
+          resources:
```

Comment:
Can we have resoures templated as well, so we can set different values for different instances?

Reply:
Sure, but afaik we need to use 443 port for service regardless the configuration because admission webhook has to be accessible from https. And there is really no point in changing the container port as it should be only specific for that pod.

Reply:
lol, the comment was only about resources, not ports. In deployment.yaml we have hardcoded request and limits. We should be able to change that per instance.

Reply:
Oh, GitHub definitely shows the ports on the highlighted comment. I'm doing it rn.